product_weight_through_uom/models/product_template.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2018 Tecnativa S.L. - David Vidal
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from odoo.addons import decimal_precision as dp
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    extra_weight = fields.Float(
```

Comment:
Put this field at product.product level, not here.

Reply:
What happens if there are not variants enabled. The products are edited through the `product.template` model and the proper values are set to their `product.product` counterpart: https://github.com/odoo/odoo/blob/11.0/addons/product/models/product_template.py#L243

Then maybe the way is to use a similar compute/inverse mechanism with this field.

Reply:
Yes, good idea.